Detections/OfficeActivity/Office_MailForwarding.yaml

```diff
@@ -24,25 +24,41 @@ query: |
   OfficeActivity
   | where TimeGenerated >= ago(timeframe)
   | where Operation =~ "Set-Mailbox"
-  | where Parameters contains "ForwardingSmtpAddress"
+  | where Parameters has "ForwardingSmtpAddress"
   | extend parsed = parse_json(Parameters)
-  | extend parameterName = parsed[1].Name, fwdingDestination = tostring(parsed[1].Value)
+  | mv-expand parsed
+  | where parsed.Name == "ForwardingSmtpAddress"
+  | extend parameterName = tostring(parsed.Name), fwdingDestination = tostring(parsed.Value)
   | where isnotempty(fwdingDestination)
   | extend ClientIPOnly = case( 
-  ClientIP has ".", tostring(split(ClientIP,":")[0]), 
-  ClientIP has "[", tostring(trim_start(@'[[]',tostring(split(ClientIP,"]")[0]))),
+  ClientIP has "." and ClientIP has ':', tostring(split(ClientIP,":")[0]), 
+  ClientIP has "." and ClientIP has '-', tostring(split(ClientIP,"-")[0]), 
+  ClientIP has ']-', tostring(trim_start(@'[[]',tostring(split(ClientIP,"]")[0]))),
+  ClientIP has ']:', tostring(trim_start(@'[[]',tostring(split(ClientIP,"]")[0]))),
+  isempty(ClientIP) and ClientIP_ has "." and ClientIP_ has ':', tostring(split(ClientIP_,":")[0]), 
+  isempty(ClientIP) and ClientIP_ has "." and ClientIP_ has '-', tostring(split(ClientIP_,"-")[0]), 
+  isempty(ClientIP) and ClientIP_ has ']-', tostring(trim_start(@'[[]',tostring(split(ClientIP_,"]")[0]))),
+  isempty(ClientIP) and ClientIP_ has ']:', tostring(trim_start(@'[[]',tostring(split(ClientIP_,"]")[0]))),
   ClientIP
   )  
   | extend Port = case(
-  ClientIP has ".", (split(ClientIP,":")[1]),
-  ClientIP has "[", tostring(split(ClientIP,"]:")[1]),
-  ClientIP
+  ClientIP has "." and ClientIP has ':', tostring(split(ClientIP,":")[1]), 
+  ClientIP has "." and ClientIP has '-', tostring(split(ClientIP,"-")[1]), 
+  ClientIP has ']-', tostring(split(ClientIP,"]-")[1]), 
+  ClientIP has ']:', tostring(split(ClientIP,"]:")[1]), 
+  isempty(ClientIP) and ClientIP_ has "." and ClientIP_ has ':', tostring(split(ClientIP_,":")[1]), 
+  isempty(ClientIP) and ClientIP_ has "." and ClientIP_ has '-', tostring(split(ClientIP_,"-")[1]), 
+  isempty(ClientIP) and ClientIP_ has ']-', tostring(split(ClientIP_,"]-")[1]),
+  isempty(ClientIP) and ClientIP_ has ']:', tostring(split(ClientIP_,"]:")[1]),
+  isnotempty(ClientIP), ClientIP,
```

Comment:
Do you want to do this check of ClientIP under case for ClientIP so value IP not available gets assigned to ClientIP instead of Port ? not much difference in functionality.

Reply:
Good catch, done.